### PR TITLE
Issue132 - Improve UI/UX of editting a verbal autopsy record

### LINKED ISSUE #132: Improve UX of VA forms with model-based fields, better navigation, hierarchy, etc
**Is your feature request related to a problem? Please describe.**
As a user reading or updating a Verbal Autopsy, I am frustrated at the UX by things like:
- Text fields showing for all answers, regardless of model (ie Textfield instead of Yes/No radio button)
- Some missing/unexplained form fields (ie. Id10012 | Id10012 instead of Id10012 | "Date of Survey" or similar)

**Describe the solution you'd like**
I would like to see the UX of the form improve via things like:
- Displaying model-based form fields based on the possible answers (date selectors, radio buttons, etc)
- Meaningful value for for every field question

### PR DESCRIPTION
**Pull request by HarunFeraidon**
closes #132 

### Motivation: 
The UI/UX of editing a record as it stands has room for improvement. For starters, every single field has a text field as input. This is a hinderance as many of the questions are not open ended and require a specific answer choice from a select set of choices.

### How I found the which model questions needed a radio field:
I did not go in and read every field. I went into an example VA found in our resources and that for each question, there was a specified "type" of answer. This includes integer, text, and the one I was looking for **"select_one."** Many of these fields had `select_one YES_NO_DK_REF` or something similar, so I used this as a guide of which fields needed a radio field.

To get every model field variable name, I did not copy and paste every single variable name. Because there were so many, I wrote a short python script where I inputted all the lines of code where we created fields for our VerbalAutopsy model. What I expected as an output, was a clean list of all our of field variable names. 
For here, I did start working by hand, where I would look at the excel sheet and find which fields had `select_one [x]`, and I would append it to a dictionary in `models.py` called RADIO_CHOICES where I map it to the choice set it needs.
 
### Room for improvement after this PR:

- There are still sets of choices besides "YES_NO_DK_REF" that are mostly used only once or a few times. I only included one, `MALE_FEMALE` because it was a small choice set. The problem with other sets is they can be large and I wasn't sure what possible choices are in the set, which is why I left them out. For example, for the question "Where did the deceased die?", the VA type expects `select_one select_18`, which is presumably a choice set of 18 types of locations.
- Some model questions require integers as input, so some of these fields should use `IntegerField` instead of `CharField`
- The UI is clean, but can probably be made to look nicer. This task would take a lot of time of designing and implementing to get to a good point
- Field value placeholders were done on there own somehow (probably some Django field default behavior). And because of [this PR code change](https://github.com/VA-Explorer/va_explorer/pull/184), sometimes the field shown on the view template ( (`show/#`), does not match the text field placeholder in the edit page (`edit/#`). This is because the implementation of view page applies a filter to some text results, while the edit page seems to automatically create text field placeholders by pulling the actual value from the DB.
An example of this is that `nan` will appear on some field placeholders, even though we got rid of instances of `nan` in the VA show page because of its unclearness.

### Testing instructions:

1.) Load up data_management tab with mock data installed.
2.) Click "View" for any row of data.
3.) click "Edit Record"
4.) Observe the UI/UX changes - addition of radio fields
5.) Make a change you'd like, then navigate to bottom of change to save that change.
6.) You should be redirected back to that VA record's view, and the changes should be present.

### CODE REVIEW

**Review by Tyler-Manderfield — COMMENTED** [review]
This PR was a very clean implementation of fixes to this issue! Specifically, I really liked how you created the constants file to keep things organized and how that constants file itself was well structured. It was very easy to follow and very modular, which is awesome. I especially liked how you created the entries by adding in the custom elements alongside the standardized variables of _yes_no_dk_ref, _ref, _dk_ref, and _other. There was no redundancy despite the loads of information that needed to be represented in this constants file. You’ll see that I have very few comments about the actual code you’ve written because everything was so seamless. The UI looks clean and the dropdowns, radio buttons and checkboxes make editing a record much easier to use and reduce the opportunity for input errors going forward. Great work!

The only comment I do have is about how the fields are updated under the hood. There seemed to be interesting behavior present when I checked the “Change History” tab after making an edit to a VA record. For the first VA that appears when I use the “mock_data_2.csv” file from Gitlab, I clicked edit, and then just scrolled to the bottom and hit save without changing anything(since I didn’t make changes to the fields, I was expecting the fields to be exactly the same as before). When I checked the change log, I noticed a few things. First, a lot of fields that started as “nan” were deleted. I didn’t paste them below because there were many. This might be expected behavior to deal with NaN/don’t knows/etc. as this PR references a previous PR in its comment to handle those, however I didn’t want to make assumptions (which is why I am noting it here). Second, and something slightly more concerning, there were a few fields that started as "nan" but then suddenly adopted new values, seemingly out of nowhere. I am guessing this is because it does not know how to map from "nan" to any of the values in the dropdown list. It seems like it might just be automatically selecting the first item in the dropdown. Here is a list of the entries that I noticed changed value for this first VA record:

Id10059 changed from nan to single
Id10063 changed from nan to no_formal_education
Id10065 changed from nan to mainly_unemployed
Id10080 changed from nan to pedestrian
Id10081 changed from nan to pedestrian
Id10088 changed from nan to dog
Id10337 changed from nan to hospital
Id10339 changed from nan to Doctor
Id10360 changed from nan to hospital
…
Id10433 changed from homeopath to ['homeopath']

This last one (Id10433) appears to be expected behavior based on what was described at the demo. However, I noticed that when I clicked edit and scrolled to the bottom again to save without actually making changes a second time, there was an additional change to this field (see below):

Id10433 had value ['homeopath'] deleted

When I clicked edit on this VA for the third time, the box for Homeopath is no longer selected. This field saving issue might be something to take a look into to ensure consistency with how the fields were functioning before.

File: va_explorer/va_data_management/constants.py [hunk 1]
```diff
@@ -0,0 +1,861 @@
...
+    "area",             # TODO: convert to dynamic dropdown
+    "hospital"          # TODO: convert to dynamic dropdown
+    "Id10007",
+    "Id10010",
+    "Id10010c",
+    "Id10017",
+    "Id10018",
+    "Id10053",
+    # "Id10054",        # textarea
+    # "Id10055",        # textarea
```
[the remark this answers is not in the document]
Comment (HarunFeraidon) [reply va_explorer/va_data_management/constants.py:613]: textfield = 1 line field
textarea = multi line field
I believe @ajbarnes left these comments, to me they look like they can be deleted.

[the remark this answers is not in the document]
Comment (HarunFeraidon) [reply va_explorer/va_data_management/constants.py:613]: I think we should keep these comments actually, could help with somebody reading the code in the future wondering what is happening to these variables.

File: va_explorer/va_analytics/views.py [hunk 2]
```diff
@@ -12,7 +12,6 @@
 from va_explorer.users.models import User
 from va_explorer.utils.mixins import CustomAuthMixin
 from va_explorer.va_analytics.filters import SupervisionFilter
-from va_explorer.va_data_management.models import PII_FIELDS, REDACTED_STRING, Location
```
[the remark this answers is not in the document]
Comment (HarunFeraidon) [reply va_explorer/va_analytics/views.py:15]: no, variables not used within file

**Comment by HarunFeraidon** [comment]
So there were three shortcomings with this PR:

1.) deletion of `nan's` and `dk's`: deletion of these values are okay for now, especially because nan's are filled in when use leaves them blank.
2.) replacement of  `nan's` and `dk's`: these values would be replaced by the first option in a dropdown. I made a hacky fix by changing the first options of every dropdown into "Don't know". Because of limited time, I think this is an okay compromise.
3.) `CheckboxSelectMultiple` unfortunately do not automatically have the values selected when the form is created. I searched for a way to fix this, but couldn't find one at the moment. Again, for limited time purposes, I will make a ticket to fix this.

**Review by Tyler-Manderfield — APPROVED** [review]
With the follow-on issue created, this PR is ready to be merged.

**Pull request merged**